### DISCUSSION TITLE
VPN access

### DISCUSSION THREAD

**Discussion by theobouwman** [caprover/caprover#1656] [Q&A]
I have a DO droplet running the CapRover manager node. It is connected to my Tailscape VPN which has an extra docker worker node attached to the main node through the VPN.

I have a public api which was easy to setup with CapRover. But how can I make a service only accessible when accessed through the Tailscale VPN?

**Comment by maietta** [comment]
That is beyond the scope of CapRover.

**Reply by githubsaturn** [reply]
@maietta is correct. CapRover does not provide such functionality. If you want to limit the incoming traffic to your service, there are two ways to do this:
- Use iptable or nfw to completely blocking the server's traffic unallowed IPs
- Edit nginx configuration in CapRover and add `allow` command to your nginx config

Note that both of these approaches are pretty advanced and you can break your instance if you are not familiar with their underlying effect.
